google/cloud/bigquery/v2/minimal/internal/json_utils.cc

```diff
@@ -85,6 +85,32 @@ void ToJson(std::chrono::system_clock::time_point const& field,
   j[name] = std::to_string(m);
 }
 
+nlohmann::json RemoveJsonKeysAndEmptyFields(std::string const& json_payload,
+                                            std::vector<std::string> keys) {
```

Comment:
Hm so we are passing these keys in by value, but using them as a reference.

The time complexity is O(N*M) where N is number of keys, and M is number of fields.

I do not know how many keys you plan to skip (and whether they show up multiple times or not).

If the keys only show up once and you know their order, you could use something like a stack, where you discard the top element once you have found it. That would only be O(M).

Reply:
Is N like... 2 or 3?